include/seqan3/utility/views/zip.hpp

```diff
@@ -6,26 +6,481 @@
+
+    sentinel() = default;
+    constexpr sentinel(sentinel<!Const> i)
+        requires Const && (std::convertible_to<std::ranges::sentinel_t<Views>,
+                                               std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>> && ...)
+    : end_(std::move(i.end_))
+    {}
+
+    template <bool OtherConst>
+        requires (std::sentinel_for<std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>,
+                                    std::ranges::iterator_t<view_helper::maybe_const<OtherConst, Views>>> && ...)
+    friend constexpr bool operator==(iterator<OtherConst> const & x, sentinel const & y)
+    {
+        return [&] <size_t... N> (std::integer_sequence<size_t, N...>)
+        {
+            return ((std::get<N>(x.current_) == std::get<N>(y.end_)) || ...);
+        } (std::index_sequence_for<Views...>{});
+
+    }
+
+    template <bool OtherConst>
+        requires (std::sized_sentinel_for<std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>,
+                                          std::ranges::iterator_t<view_helper::maybe_const<OtherConst, Views>>> && ...)
+    friend constexpr std::common_type_t<std::ranges::range_difference_t<view_helper::maybe_const<OtherConst, Views>>...>
+        operator-(iterator<OtherConst> const & x, sentinel const & y)
+    {
+        return [&] <size_t... N> (std::integer_sequence<size_t, N...>)
+        {
+            return std::ranges::min({(zip::abs(std::get<N>(x.current_) - std::get<N>(y.end_)), ...)});
+        } (std::index_sequence_for<Views...>{});
+    }
+
+    template <bool OtherConst>
+        requires (std::sized_sentinel_for<std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>,
+                  std::ranges::iterator_t<view_helper::maybe_const<OtherConst, Views>>> && ...)
+    friend constexpr std::common_type_t<std::ranges::range_difference_t<view_helper::maybe_const<OtherConst, Views>>...>
+        operator-(sentinel const & y, iterator<OtherConst> const & x)
+    {
+        return - (x - y);
+    }
+};
+
+struct zip_fn
+{
+    template <typename urng_t>
+    constexpr auto operator()(urng_t && rng) const
+    {
+        return adaptor_from_functor{*this, std::forward<urng_t>(rng)};
+    }
+
+    template <typename... urng_ts>
+        requires (sizeof...(urng_ts) == 0)
+    constexpr auto operator()(urng_ts &&... ranges) const
+    {
+        return std::views::empty<seqan3::common_tuple<>>;
+    }
+
+    template <typename... urng_ts>
+        requires (sizeof...(urng_ts) > 1)
+    constexpr auto operator()(urng_ts &&... ranges) const
+    {
+        return zip_view{std::forward<urng_ts>(ranges)...};
+    }
+};
+
+} // namespace seqan3::detail
+//!\endcond
 
 namespace seqan3::views
 {
 
-/*!\brief A zip view
+/*!\brief A view adaptor that produces a tuple-like value of all passed views.
  * \ingroup utility_views
- * \details
- * \noapi{This is currently range-v3's zip implementation.}
```

Comment:
The no-api tag still applies here?

Reply:
Yes, it does :)
I changed it to say something along the lines "This is an implementation of an STL feature. In due time, we will switch to std::views::zip"